src/main/java/core/basesyntax/strategy/QuantityIncrementer.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.mapvalidator.KeyPresentValidator;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public interface QuantityIncrementer extends KeyPresentValidator {
+    default void incrementByQuantity(Map<String, Integer> fruitQuantityMap,
+                                     FruitTransaction transaction) {
+        validateMap(fruitQuantityMap, transaction);
+        BiFunction<String, Integer, Integer> incrementFruitQuantityFunction =
+                (fruitName, quantity) -> quantity + transaction.getQuantity();
+        fruitQuantityMap.computeIfPresent(transaction.getFruit(), incrementFruitQuantityFunction);
```

Comment:
The method 'incrementByQuantity' should not be a default method in an interface. It's better to define it in an abstract class if you need shared code, or implement it in the concrete classes that require this behavior. Having non-abstract methods in interfaces can lead to issues with class design and understanding the responsibility of the interface.